File: 3321-type-of-triangle/3321-type-of-triangle.go
func triangleType(nums []int) string {
    a, b, c := nums[0], nums[1], nums[2]
    if a == b && b == c {
        return "equilateral" 
    }
    if (a == b && a + b > c) || (b == c && b + c > a) || (a == c && a + c > b) {
        return "isosceles"
    }
    if a + b > c && b + c > a && a + c > b {
        return "scalene"
    }
    return "none"
}

File: 3321-type-of-triangle/3321-type-of-triangle.py
class Solution:
    def triangleType(self, nums: List[int]) -> str:
        a, b, c = nums[0], nums[1], nums[2]
        if a == b and b == c:
            return "equilateral"
        if (a == b and a + b > c) or (b == c and b + c > a) or (a == c and a + c > b):
            return "isosceles"
        if a + b > c and b + c > a and a + c > b:
            return "scalene"
        return "none"